scripts/apply_clang_format.py

```diff
@@ -0,0 +1,200 @@
+        else:
+            print("Found pre-existing clang-format binary, skipping download")
+
+    # Now that the binary is where it should be, hash it.
+    actual_bin_hash = compute_file_sha1(filename)
+
+    # Load the reference hash.
+    # This dictionary maps each platform to a relative path to a file containing its reference hash.
+    PLATFORM_TO_HASH = {
+        "Darwin": os.path.join("mac", "clang-format-mojave"),
+    }
+
+    plat = platform.system()
+
+    # If the host platform is not in PLATFORM_TO_HASH, it is unsupported.
+    if plat not in PLATFORM_TO_HASH:
+        print("Unsupported platform: {}".format(plat))
+        return False
+
+    # This is the path to the file containing the reference hash.
+    hashpath = "{}/scripts/clang_format_hash/{}".format(pytorch_root, PLATFORM_TO_HASH[plat])
+
+    if not os.path.exists(hashpath):
+        print("Unable to find reference binary hash")
+        return False
+
+    # Load the reference hash and compare the actual hash to it.
+    with open(hashpath, "r") as f:
+        reference_bin_hash = f.readline()
+
+        if reference_bin_hash != actual_bin_hash:
+            print("The downloaded binary is not what was expected!")
+
+            # Err on the side of caution and try to delete the downloaded binary.
+            try:
+                os.unlink(filename)
+            finally:
+                return False
+        else:
+            # Make sure the binary is executable.
+            mode = os.stat(filename).st_mode
+            mode |= stat.S_IXUSR
+            os.chmod(filename, mode)
+
+    return True
+
+
+def validate_pytorch_root(pytorch_root):
+    """Check that pytorch_root probably is the root of a PyTorch working directory by looking for .clang-format."""
+
+    # Check if pytorch_root has a .clang-format. If it does, it's probably a viable PyTorch root?
+    clang_format_path = os.path.join(pytorch_root, ".clang-format")
+
+    if not os.path.exists(clang_format_path):
+        print("Could not find .clang-format in {}. Are you sure this is root of your PyTorch directory?".format(pytorch_root))
+        return False
+
+    return True
+
+
+def parse_args(args):
+    """Parse and return command-line arguments."""
+    parser = argparse.ArgumentParser(
+        description="Invoke clang-format to files that have been whitelisted for automatic formatting."
+    )
+    parser.add_argument(
+        "--pytorch-root",
+        required=False,
+        help="The root of your PyTorch working directory (default is current directory)",
+        default=".",
+    )
+
+    a = parser.parse_args(args)
+    return parser.parse_args(args)
+
+
+def main(args):
+    # Parse arguments.
+    options = parse_args(args)
+    # Make sure the PyTorch root is absolute.
+    pytorch_root = gcf.sanitize_path(options.pytorch_root)
+    # Check that the given PyTorch root is actually the root and change the current working directory to it.
+    ok = validate_pytorch_root(pytorch_root)
```

Comment:
Since this script is in a known location with respect to the root, might be more robust to just navigate relative to `__file__`, which is an interpreter-set variable corresponding to this file path.

Reply:
Fixed.